accel-pppd/radius/radius.c

```diff
@@ -1006,6 +1017,52 @@ static int load_config(void)
 	return 0;
 }
 
+static void ev_radius_coa_vrf(struct ev_radius_t *ev)
+{
+	struct rad_attr_t *attr;
+
+	log_ppp_debug("ev_radius_coa_vrf started\n");
+
+	list_for_each_entry(attr, &ev->request->attrs, entry) {
+		int vendor_id = attr->vendor ? attr->vendor->id : 0;
+
+		if (vendor_id != Vendor_Accel_PPP)
+			continue;
+
+		if (attr->attr->id == Accel_VRF_Name) {
+			if (attr->len < 1) {
+				log_ppp_error("ev_radius_coa_vrf vrf attr len < 1\n");
+				ev->res = -1;
+				return;
+			}
+
+			log_ppp_debug("ev_radius_coa_vrf vrf attribute found value=%s\n", attr->val.string);
+
+			if (attr->len == 1 && attr->val.string[0] == '/') {
+				log_ppp_debug("ev_radius_coa_vrf vrf removing\n");
+				if (ap_session_remove_vrf(ev->ses)) {
+					ev->res = -1;
+					return;
+				}
+				if (ev->ses->vrf_name)
+					_free(ev->ses->vrf_name);
+				ev->ses->vrf_name = NULL;
+				return;
+			}
+
+			if (ev->ses->vrf_name)
```

Comment:
let ap_session_set_vrf() handle actual vrf_name for the session inside ot, just pass received value